src/serializer/ResidualFunctions.js

```diff
@@ -117,6 +117,51 @@ export class ResidualFunctions {
     if (!this.firstFunctionUsages.has(val)) this.firstFunctionUsages.set(val, bodyReference);
   }
 
+  _shouldUseFactoryFunction(funcBody: BabelNodeBlockStatement, instances: Array<FunctionInstance>) {
+    function shouldInlineFunction(): boolean {
```

Comment:
Any reason function expression is better? Personally, I like function declaration better than function expression because the code is more readable(you can easily spot if a symbol is a function or normal variable)

Reply:
I wasn't saying use a function expression. Since the code is only called once, why is it in a function at all? putting lines 123-127 after line 132 and using the `shouldInline` boolean instead of calling shouldInlineFunction  would be clearer imo.

Reply:
Function can make code more readable because it defines a well documented scope which people can be comfortable look at its name without dig into its implementation.
I do not think creating a function makes the code unclear here.